recaptcha_enterprise/demosite/src/main/java/app/MainController.java

```diff
@@ -36,68 +39,356 @@
+  private static final Properties PROPERTIES = new Properties();
 
   static {
+    SAMPLE_THRESHOLD_SCORE = 0.50;
+
     CONTEXT.put("project_id", System.getenv("GOOGLE_CLOUD_PROJECT"));
     CONTEXT.put("site_key", System.getenv("SITE_KEY"));
+
+    // Parse property file and read available reCAPTCHA actions. All reCAPTCHA actions registered
+    // in the client should be mapped in the config file. This will be used to verify if the token
+    // obtained during assessment corresponds to the claimed action.
+    try (InputStream input = MainController.class.getClassLoader()
+        .getResourceAsStream("config.properties")) {
+      PROPERTIES.load(input);
+    } catch (Exception e) {
+      System.out.println("Exception while loading property file...");
+    }
   }
 
+  /**
+   * Return homepage template.
+   */
   @GetMapping(value = "/")
   public static ModelAndView home() {
     return new ModelAndView("home", CONTEXT);
   }
 
-  @GetMapping(value = "/store")
-  public static ModelAndView store() {
-    return new ModelAndView("store", CONTEXT);
+  /**
+   * On homepage load, execute reCAPTCHA Enterprise assessment and take action according to the
+   * score.
+   */
+  @PostMapping(value = "/on_homepage_load", produces = "application/json")
+  public static @ResponseBody ResponseEntity<HashMap<String, HashMap<String, String>>> onHomepageLoad(
+      @RequestBody Map<String, HashMap<String, String>> jsonData) {
+    final HttpHeaders httpHeaders = new HttpHeaders();
+    httpHeaders.setContentType(MediaType.APPLICATION_JSON);
+    String recaptchaAction = PROPERTIES.getProperty("recaptcha_action.home");
+    HashMap<String, HashMap<String, String>> data;
+    Assessment assessmentResponse;
+    String verdict;
+
+    try {
+      // <!-- ATTENTION: reCAPTCHA Example (Server Part 1/2) Starts -->
+      assessmentResponse = CreateAssessment.createAssessment(
+          CONTEXT.get("project_id"),
+          CONTEXT.get("site_key"),
+          jsonData.get("recaptcha_cred").get("token"));
+
+      // Check if the token is valid, score is above threshold score and the action equals expected.
+      if (assessmentResponse.getTokenProperties().getValid() &&
+          assessmentResponse.getRiskAnalysis().getScore() > SAMPLE_THRESHOLD_SCORE &&
+          assessmentResponse.getTokenProperties().getAction().equals(recaptchaAction)) {
+        // Load the home page.
+        // Business logic.
+        // Classify the action as not bad.
+        verdict = "Not Bad";
+      } else {
+        // If any of the above condition fails, trigger email/ phone verification flow.
+        // Classify the action as bad.
+        verdict = "Bad";
+      }
+      // <!-- ATTENTION: reCAPTCHA Example (Server Part 1/2) Ends -->
+
+      // Return the risk score.
+      HashMap<String, String> result = new HashMap<>() {{
+        put("score", String.valueOf(assessmentResponse.getRiskAnalysis().getScore()));
+        put("verdict", verdict);
```

Comment:
Should these be inline? Can we split them to be seperate operation for clarity?

Reply:
ditto